feat!: assistive-pricing as a Worker

gentlementlegen opened the pull request:
Resolves #5 

<!--
- You must link the issue number e.g. "Resolves #1234"
- Please do not replace the keyword "Resolves" https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

gentlementlegen commented:
## Major changes

- `assistive-worker` now running as a Worker
- fixed Knip and reported errors
- related tests and Jest running at `pull_request`
- fixed command error being displayed all the time even for unrelated commands

gitcoindev commented:
Hi @gentlementlegen there is a small commit needed for knip workflow that removes superflous steps. See comment from @rndquu  https://github.com/ubiquity/rpc-handler/pull/13#issuecomment-2123006059 . This does not touch any of your changes, I will add this commit to the pull request.

gitcoindev commented:
> Hi @gentlementlegen there is a small commit needed for knip workflow that removes superflous steps. See comment from @rndquu [ubiquity/rpc-handler#13 (comment)](https://github.com/ubiquity/rpc-handler/pull/13#issuecomment-2123006059) . This does not touch any of your changes, I will add this commit to the pull request.

My bad - I checked and you pulled latest changes already, thank you!

gitcoindev commented:
@gentlementlegen I checked out the pr workspace, and I am able to test locally with jest.
Would you mind adding to `README.md` steps needed to execute and test the worker locally including supabase setup, if needed? I set supabase `.dev.vars` to my instance but I see `{"error":"Error: null is not a valid content type"}` when I connect to worker started with `yarn worker` command and I am not sure what I am missing and what to expect. Perhaps adding this info would make life easier to anyone who would work on the plugin afterwards. I also see cypress is still a placeholder, will it be used in this repository?

gentlementlegen commented:
@gitcoindev Thanks for the feedback. I removed Cypress, and added instructions in the README plus a `.http` test file. Let me know if that is clear enough or if I should ass more instructions.

gitcoindev commented:
@gentlementlegen thank you for adding the example, I am approving the pull request from my side.

whilefoo commented on src/worker.ts:
```diff
@@ -0,0 +1,32 @@
+  async fetch(request: Request, env: Env): Promise<Response> {
+    try {
+      const contentType = request.headers.get("content-type");
+      if (contentType !== "application/json") {
```
also check that it's a POST method

whilefoo commented on src/shared/pricing.ts:
```diff
@@ -42,20 +41,3 @@ export function calculateLabelValue(label: string): number {
   return 0;
 }
-
-export function calculateDurations(labels: Label[]): number[] {
```
why was this removed?

gentlementlegen replied:
This was to satisfy Knip because this export wasn't used anywhere

whilefoo replied:
hmm that's weird

gentlementlegen replied:
Added it back as a comment. If it is supposed to be used somewhere let me know I can have a look.
